extension/WebJobs.Extensions.RabbitMQ/Trigger/RabbitMQListener.cs

```diff
@@ -152,6 +160,32 @@ public ScaleStatus GetScaleStatus(ScaleStatusContext<RabbitMQTriggerMetrics> con
             return this.GetScaleStatusCore(context.WorkerCount, context.Metrics?.ToArray());
         }
 
+        internal static Activity StartActivity(BasicDeliverEventArgs ea)
+        {
+            Activity activity;
+            if (ea.BasicProperties.Headers != null && ea.BasicProperties.Headers.ContainsKey("traceparent"))
+            {
+                // check if traceId present in header
+                byte[] traceParentIdInBytes = ea.BasicProperties.Headers["traceparent"] as byte[];
+                string traceparentId = Encoding.Default.GetString(traceParentIdInBytes);
+                activity = Source.StartActivity("RabbitMQ.function.trigger", ActivityKind.Consumer, traceparentId);
+            }
+            else
+            {
+                // create new traceId, StartActivity will create and start new activity
+                activity = Source.StartActivity("RabbitMQ.function.trigger", ActivityKind.Server);
+                if (ea.BasicProperties.Headers == null)
+                {
+                    ea.BasicProperties.Headers = new Dictionary<string, object>();
+                }
+
+                byte[] traceParentIdInBytes = Encoding.Default.GetBytes(activity?.Id);
```

Comment:
Change to `activity.Id` as it gives impression that `activity` can possibly be `null`.